main.py

```diff
@@ -742,22 +742,29 @@ def make_move(move, board:Board):
     #update king position
     if board.board[position] == K or board.board[position] == k:
         board.king_position[board.side] = target
-    else:
+    #update other pieces positions
+    elif board.board[position] != K or board.board[position] != k:
         if char_ascii[board.board[position]] == '.': 
             board.undo_move()
```

Comment:
the move didn't get executed so far because you moved the make move function down

Reply:
you are right. I was panicking and triple checked everything. 748 should be deleted